File: to-do-app/to-do-go/src/controller/user_controller.go
package controller

import (
	"encoding/json"
	"errors"
	"io/ioutil"
	"net/http"

	"renanlelis.github.io/portfolio/to-do-go/src/business"
	"renanlelis.github.io/portfolio/to-do-go/src/business/messages"
	"renanlelis.github.io/portfolio/to-do-go/src/controller/form"
	"renanlelis.github.io/portfolio/to-do-go/src/controller/response"
	"renanlelis.github.io/portfolio/to-do-go/src/security"
)

// UpdateUserProfile update user information like first name or last name
func UpdateUserProfile(w http.ResponseWriter, r *http.Request) {
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusInternalServerError, errors.New(messages.GetErrorMessage()), r)
		return
	}
	var form form.UserProfileForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}
	bErr := business.UpdateUserProfile(userID, form.FirstName, form.LastName)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

// UpdateUserPassword update user password after user is logged in
func UpdateUserPassword(w http.ResponseWriter, r *http.Request) {
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		response.Err(w, http.StatusUnprocessableEntity, err, r)
		return
	}
	userID, err := security.GetUserID(r)
	if err != nil {
		response.Err(w, http.StatusInternalServerError, errors.New(messages.GetErrorMessage()), r)
		return
	}
	var form form.UserPasswordForm
	if err = json.Unmarshal(reqBody, &form); err != nil {
		response.Err(w, http.StatusBadRequest, err, r)
		return
	}

	bErr := business.UpdateUserPassword(userID, form.Password)
	if bErr.Status > 0 {
		response.Err(w, bErr.Status, errors.New(bErr.ErrorMessage), r)
		return
	}
	response.JSON(w, http.StatusOK, nil, r)
}

File: to-do-app/to-do-node/src/controller/userController.ts
import { Request, Response } from "express"
import msg from '../business/messages/messages';
import { Error, AppErrorType } from "../business/errorType";
import userService from '../business/userService';
import token from '../business/token/token'

const updateUserProfile = (req: Request, res: Response) => {
    if (!req.body) {
        return res.status(AppErrorType.INVALID_INPUT).json({
            errorMessage: msg.getErrorMessageInputValues()
        });
    }
    const userID = token.getUserID(req)
    userService.updateUserProfile(userID, req.body.firstName, req.body.lastName)
        .then(() => res.status(200).send())
        .catch(err => {
            if (err instanceof Error) return res.status(err.errorType).json({ errorMessage: err.errorMessage })
            else return res.status(AppErrorType.INTERN_ERROR).json({ errorMessage: msg.getErrorMessage() })
        })
}

const updateUserPassword = (req: Request, res: Response) => {
    if (!req.body) {
        return res.status(AppErrorType.INVALID_INPUT).json({
            errorMessage: msg.getErrorMessageInputValues()
        });
    }
    const userID = token.getUserID(req)
    userService.updateUserPassword(userID, req.body.password)
        .then(() => res.status(200).send())
        .catch(err => {
            if (err instanceof Error) return res.status(err.errorType).json({ errorMessage: err.errorMessage })
            else return res.status(AppErrorType.INTERN_ERROR).json({ errorMessage: msg.getErrorMessage() })
        })
}


export default {
    updateUserProfile,
    updateUserPassword
}